cli/command/container/utils.go

```diff
@@ -53,27 +75,20 @@ func waitExitOrRemoved(ctx context.Context, dockerCli *command.DockerCli, contai
 		return false
 	}
 
-	// Get events via Events API
-	f := filters.NewArgs()
-	f.Add("type", "container")
-	f.Add("container", containerID)
-	options := types.EventsOptions{
-		Filters: f,
-	}
-
-	eventCtx, cancel := context.WithCancel(ctx)
-	eventq, errq := dockerCli.Client().Events(eventCtx, options)
-
 	go func() {
 		defer cancel()
 
 		for {
 			select {
+			case <-eventCtx.Done():
+				if removeErr != nil {
+					logrus.Errorf("error removing container: %v", removeErr)
+					return
+				}
```

Comment:
I see. This is a little funky sense we're sending the exitCode in the even processor above in the everything is ok case.
Should probably be in a defer or something, WDYT?